internal/pkg/cli/storage_init.go

```diff
@@ -318,9 +303,8 @@ https://aws.github.io/copilot-cli/docs/developing/additional-aws-resources/#what
 
 func (o *initStorageOpts) askStorageType() error {
 	if o.storageType != "" {
-		return nil
+		return o.validateStorageType()
```

Comment:
One last request: maybe we should add a comment here explaining why we don't `return nil` instead, which is our common pattern for `Ask()`.